docs.md

```diff
@@ -1,23 +1,216 @@
 ## Full documentation of the API
 
-### Architecture diagram of the API [Here](https://www.figma.com/file/684S7kO4dPQbbZFZr6xZOn/Rego-builder?node-id=0%3A1)
+
+
+
+
+Table of Contents
+=================
+
+  * [Introduction](#introduction)
+  * [Features](#features)
+  * [How the policy manager works](#architecture)
+  * [Installation](#installation)
+  * [Usage](#usage)
+    1. [Input_props_equals](#input_props_equals)
+         * [Handling * as the wildcard flag](#Handling-'*'-as-the-wildcard-flag)
+         * [Allowing all path parameter after a path section, except one](#Handling-'*'-as-the-wildcard-flag)
+         * [ Handling equality check between a particular property on the request object and a valueHandling * as the wildcard flag](#Handling-'*'-as-the-wildcard-flag)
+         * [Allow access to a particular path](#allow-access-to-a-particular-path)
+      2. [Input_props_in](#input_props_in)
+      3. [Input_props_in_as](#input_props_in_as)
+      4. [allow_full_access](#allow_full_access)
+
+
+
+
+
+  Introduction
+============
+This API is responsible for converting JSON variables to REGO which in turn is used in application policy management. The policy management application is born from the need to manage Open Policy Agent ( OPA ) policies written in REGO from a user interface for non-technical users and an API through JSON for technical users.
+
+  Features
+============
+   - Translating JSON to REGO
+   - CRUD operations for policies for persistence.
+   - Supports existing policies for editing.
+   - Supports creating new policies from scratch.
+
+   Architecture
+   ============
+   The architecture of the API is based on the following principles:
+   - The API is written in Python and FastAPI framework.
+   - The API is written in a way that it is easy to use, extend, test and maintain.
 
 ![Screenshot 2022-06-13 at 09 55 43](https://user-images.githubusercontent.com/67229938/173343113-d51d72b4-84c8-4c3b-8555-af41e59cd2de.png)
 
+View the Architecture in the browser [Here](https://www.figma.com/file/684S7kO4dPQbbZFZr6xZOn/Rego-builder?node-id=0%3A1) 
+
+
+
+   Installation
+   ============
+- Manual installation - Available [Here](https://github.com/r-scheele/rego_builder#readme)
+
+- Build your own image and start the container -
+   ```bash
+   $ Configure environment variables - the .env file
+   $ docker-compose -f docker-compose.dev.yml up -d
+
+   ```
+- Pull the official image from Docker Hub - and start the container 
+
+   ```bash
+   $ docker pull rscheele3214/rego-builder:latest
+   $ docker run rscheele3214/rego-builder:latest
+   ```
+
+
+Usage
```

Comment:
Before directly going into usage, I'll expect to be briefed on how the translation from JSON to REGO works. I have no idea of commands - what do they stand for as well?

I'll expect a brief/detailed section of what commands are - you can then list the various commands here.

After explaining what the commands are, you should give us a lead-in on how they are compiled to form a rule - command, and properties. There's no explanation of what properties are.

Reply:
Alright would submit another commit for review soon